doc/builders/special/darwin-builder.section.md

```diff
@@ -61,3 +61,74 @@ builders-use-substitutes = true
 ```ShellSession
 $ sudo launchctl kickstart -k system/org.nixos.nix-daemon
 ```
+
+## Example flake usage
+
+```
+{
+  inputs = {
+    nixpkgs.url = "path:/Users/geraint.ballinger/Documents/personal/nixpkgs";
+    nixpkgs-unstable.url = github:NixOS/nixpkgs/nixpkgs-unstable;
+    darwin.url = "github:lnl7/nix-darwin/master";
+    darwin.inputs.nixpkgs.follows = "nixpkgs-unstable";
+  };
+
+  outputs = { self, darwin, nixpkgs, ... }@inputs:
+  let
+
+    inherit (darwin.lib) darwinSystem;
+    pkgs = nixpkgs.legacyPackages.aarch64-darwin;
+    system = pkgs.stdenv.hostPlatform.system;
+    linuxSystem = builtins.replaceStrings [ "darwin" ] [ "linux" ] system;
+
+    darwin-builder = import "${nixpkgs}/nixos" {
+      system = linuxSystem;
+      configuration = {
+        imports = [
+          "${nixpkgs}/nixos/modules/profiles/macos-builder.nix"
+        ];
+        virtualisation.host.pkgs = pkgs;
+      };
+    };
```

Comment:
```suggestion
    darwin-builder = nixpkgs.nixosSystem {
      system = linuxSystem;
      modules = [
        "${nixpkgs}/nixos/modules/profiles/macos-builder.nix"
        { virtualisation.host.pkgs = pkgs; }
      ];
    };
```

Reply:
I just tried this and it causes an error because the system.nixos.revision parameter gets set to self.rev by the flake. I think either we have to revert back to the original ,or else sacrifice `system.nixos.revision = null`. What are your thoughts?

Reply:
i think this is fine for now